suave/builder/api/api_server.go

```diff
@@ -0,0 +1,43 @@
+package api
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// sessionManager is the backend that manages the session state of the builder API.
+type sessionManager interface {
+	NewSession() (string, error)
+	AddTransaction(sessionId string, tx *types.Transaction) (*types.SimulateTransactionResult, error)
+}
+
+func NewServer(s sessionManager) *Server {
+	api := &Server{
+		sessionMngr: s,
+	}
+	return api
+}
+
+type Server struct {
+	sessionMngr sessionManager
```

Comment:
should we have a mutex on this? seems possible you could call start a session from another SUAPP before the last new session call has been completed?

Reply:
The locks are on `sessionManager`.